### DISCUSSION TITLE
osu! Story Mode

### DISCUSSION THREAD

**Discussion by Rekunan** [ppy/osu#22270] [Ideas] [answered]
To preface, a lot of this is taken from Arcaea's course mode and story, as well as dan courses in general.

Base idea: An osu! Story Mode that contains story and gameplay, served through stages, with the ability to choose what mode(s) you want to play.

Story (rough idea):
You follow the life of _insert mode mascot(s) here (combination if ur doing multiple modes)_ as they venture through the world of Lazer, guiding them as they fight enemies and solve puzzles on their journey to discovering the big mystery.

Gameplay:
In each stage is a set of maps that you have to obtain at least a certain acc/pp/score/whatever in order to progress through the stage. The final boss of each stage would be all the maps together, but you're forced to play them one after another in one sitting. After completing a stage, you're allowed to move on to the next, as well as retry any past stages at any time you wish. As you progress through stages, they increase in difficulty.

Both:
Enemies in the story are themed/designed based on the maps in the corresponding or preceding stage.

Other:
There could also be tiered rewards for completing stages, such as an achievement medal that would increase in quality as you set better acc/pp/score/whatever on the stage.

My main want is the stages, if needed the story can be saved for later.

**Comment by peppy** [comment]
I've been saying for a decade that this would be awesome, but it requires someone to make it. The systems already exist to make storyboards, but someone has to actually get their shit together and make a campaign. I'm very sad this hasn't happened yet in 15 years (there was much more interest and even some initial progress in the early days, but these days it's only PP this PP that).

**Comment by Biokax** [comment]
I really like the idea of creating a story mode for osu! It will make the game more lively and allow new players to learn how to play in a different way than just a small tutorial map giving the bare minimum of information about the game, and also giving the mascots something else than just appearing on fan arts.

It can allow to introduce different mechanics of the game in a given order without forcing players to read the wiki or watching youtube tutorials. Like for example the introduction of new patterns, playing with two keys, the importance of the different statistics of a map, etc.

To make things happen, I think we should first communicate with the community about the desire to make a story mod, and the need to find motivated people. I'm thinking of a community contest to create a scenario in several chapters introducing the different mascots of the game, with mechanics related to the game mode.

Once the most interesting scenario is chosen, I think it will be necessary to develop in-game elements to make the mascots appear with a dialog field on the game screen, and make this tool available to the community in a new contest by asking to create the first part of the scenario by combining the mapping/storyboard/ and the use of the tool to display the mascots in order to make a level coherent with what we want of the story mode.

With the results of the contests, we can get an idea of where we want to go, and eventually look for motivated community members to make a complete story. I know it's not optimal to have a contest just to motivate players to create a story mode, but I think it's necessary to get the word out to as many players as possible on the subject.

It will also allow to show to everyone on the home screen of the game that there is a story mode in progress.

**Reply by Rekunan** [reply]
Funny that you mention motivation, because I already got people dming and pinging me about wanting to help out with this idea, even one who's already made contributions to Lazer itself. I do plan now to organize something for this sometime in the future and make it public. Considering how hard the devs are working on Lazer, I think it's better if I, and any others willing to help, manage the PR (such as polls) to lessen their load, perhaps only coming in when we have a close to, if not finished product.

I do agree with the idea of introducing game mechanics through this mode, especially niche QOL ones such as keyboard shortcuts. Once I get around to organizing this, there'll def be lots of development and growth in terms of ideas.

**Reply by Biokax** [reply]
The story mode indeed isn't a priority for the devs and that cool if people are motivated to help, but i think it need more official communication in order to get more people involved. What about adding a new channel to the osu!dev discord dedicated to the osu!story ?

**Reply by mk56-spn** [reply]
> The story mode indeed isn't a priority for the devs and that cool if people are motivated to help, but i think it need more official communication in order to get more people involved. What about adding a new channel to the osu!dev discord dedicated to the osu!story ? 

This'll become a cluste****  in 10 minutes unless people are vetted before joining. At which point you are back to the issue of burdening the developers with the project

**Reply by Walavouchey** [reply]
something like this will need organisation on a larger scale, so even if progress is communicated in osu!dev, a separate server is most definitely necessary

**Reply by peppy** [reply]
I would recommend organising a small team and not expanding too big. It will not work unless you have a core group of <5 people making it happen.

**Reply by Rekunan** [reply]
Good point yeah, luckily I think a good core has already formed, so we can start with that for now, and then expand/publicize as needed.

**Comment by Corne2Plum3** [comment]
I personally would love to see a story mode and I was also thinking how it would be implemented. Especially the maps we're going to play. The song would be only from Featured Artists, and each song would have 3 or 4 difficulties, like Arcaea or Cytus, so player form any level could play it. For STD mode I suggest the following Star Rates (it's an example and in some cases it's nice to go out of range):

* Past: <2.70*
* Present: 2.70 - 4.70*
* Future: 4.70 - 6.70*
* Beyond (if in some songs you want to put very difficult maps): >6.70*

I picked the difficulties names from Arcaea in the example above but you get the idea. I don't think using the same difficulties naming system we already have would be a good idea as there would be missing and inconsistent difficulties if we have only 3 difficulties for each song (e.g. Easy/Hard/Expert in one song, Normal/Hard/Insane in another, etc.). If you have very good and original names for the 3 (4) difficulties, it would be nice.

**Reply by mk56-spn** [reply]
There's already plans heavily underway

**Reply by Rekunan** [reply]
Adding on, some of these ideas we do agree with and will probably use, such as only Featured Artists. And to not reveal too much, there's probably also going to be some ideas from my original discussion post that're scrapped depending on how the team progresses.

**Comment by Rekunan** [comment, marked as answer]
Core team has been formed and work has started.

**Reply by Sr3qt** [reply]
Are there any progress updates? How is the development going?

**Reply by LFACC0** [reply]
Hi! how can I contact that team? I would love to see the material and ideas they have at this moment, I'm trying to start something like a source book or similar and try to align this to their work would be the better way to begin with

**Reply by I-AmBen** [reply]
Any update? Any progress? Where an I see the progress?

**Reply by LFACC0** [reply]
@I-AmBen , Currently the leader of the project is busy with other stuff until the beginning of the next year. Right now he is brainstorming ideas and slowly choosing potential key members for the different tasks required in order to make this happen!

Stay tuned!

**Reply by Givikap120** [reply]
@LFACC0 who's the leader of the project and how could I contact them?

**Comment by deniz-blue** [comment]
hello yes i would like to give some stupid ideas

> Story (rough idea):
> You follow the life of insert mode mascot(s) here (combination if ur doing multiple modes) as they venture through the world of Lazer, guiding them as they fight enemies and solve puzzles on their journey to discovering the big mystery.

In my opinion, this is bad. The gameplay idea is good but the story idea seems too set in stone, i think it would be better if the story wasn't centered around some mascot (an already defined character) and the generic player, like yourself. That change would make the story more centered towards the player and would make it easier for the players to project themselves or get invested in the story more. You can see this in something like Just Shapes and Beats where the character you are playing as doesnt have much already defined personality, or in Undertale where you dont have to conform to a character's personality and rules.

I agree with Biokax - teaching new players the mechanics of the game via this story mode would be amazing. As for the story, in my opinion, it could have something to do with how the game works maybe? Inspired from other games (JSAB, Fez, etc) maybe the game could "break" and you would complete beatmaps to somehow help fix it? Aspire maps would be useful for this maybe?

tldr: mascot is predefined therefore it might break immersion so having the main character be yourself aka the player could be better

**Comment by Givikap120** [comment]
Make a lazer mod introduction campaign

**Comment by Sr3qt** [comment]
Does anybody know what happened to the development of story mode? I cannot seem to find any other mention of how much progress has been made. I am assuming development has stopped.

**Comment by Tenexxt** [comment]
Imagine osu! story mode will hit you the same way as Persona 3 and Cyberpunk. For a game that looks 'cute' that would be kinda funny, but i deffinetly would like it

**Comment by Sr3qt** [comment]
I dmed the devs and they confirmed that they were still working on the game, and that no public announcement is planned.

**Reply by okolenmi** [reply]
Nice. I hope it will be released before the next decade.

**Reply by Tenexxt** [reply]
> Nice. I hope it will be released before the next decade.

Don't worry, it will release before silksong 100%

**Reply by ziv-vy** [reply]
oh that didn't age so well....

**Reply by Tenexxt** [reply]
OK then we *have* to get a teaser before GTA VII atleast, right?

Right???
